material-stepper/src/main/java/com/stepstone/stepper/adapter/AbstractStepAdapter.java

```diff
@@ -1,19 +1,49 @@
 package com.stepstone.stepper.adapter;
 
 import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.View;
+import android.view.ViewGroup;
 
 import com.stepstone.stepper.Step;
 
-/**
- * Created by leonardo on 18/12/16.
- */
+import java.util.LinkedList;
+import java.util.List;
 
 /**
  * A base adapter class which returns step to use inside of the {@link com.stepstone.stepper.StepperLayout}.
  * This class is intended to be inherited if you need to use {@link com.stepstone.stepper.StepperLayout} without fragments.
- * Otherwise, you should use {@link AbstractStepAdapter}
+ * Otherwise, you should use {@link AbstractFragmentStepAdapter}
  */
-public abstract class AbstractStepAdapter<T extends Step> extends PagerAdapter implements StepAdapter<T> {
+public abstract class AbstractStepAdapter extends PagerAdapter implements StepAdapter {
+
+    private List<Step> pages = new LinkedList<>();
+
+    @Override
+    public Step findStep(ViewPager viewPager, int position) {
+        return pages.size() > 0 ? pages.get(position) : null;
+    }
+
+    @Override
+    public final View instantiateItem(ViewGroup container, int position) {
```

Comment:
I'm not sure if I'd implement this like that. I use Conductor in most of my project, so this implementation would be different, like this: [https://github.com/bluelinelabs/Conductor/blob/develop/conductor-support/src/main/java/com/bluelinelabs/conductor/support/ControllerPagerAdapter.java](https://github.com/bluelinelabs/Conductor/blob/develop/conductor-support/src/main/java/com/bluelinelabs/conductor/support/ControllerPagerAdapter.java)
Since you marked this as final, that would be an issue here. 
I'd go with the abstract here to be more flexible.